src/api/useViewportFeatures.js

```diff
@@ -15,13 +16,23 @@ export default function useViewportFeatures(
 
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const computeFeatures = useCallback(
-    debounce(({ tiles, viewport, uniqueIdProperty, sourceId }) => {
-      const features = viewportFeatures({ tiles, viewport, uniqueIdProperty });
+    debounce(async ({ tiles, viewport, uniqueIdProperty, sourceId }) => {
+      const tilesCleaned = tiles.map(({ data, isVisible, bbox }) => ({
+        data,
+        isVisible,
+        bbox
+      }));
+
+      await executeTask(sourceId, Methods.VIEWPORT_FEATURES, {
+        tiles: tilesCleaned,
+        viewport,
+        uniqueIdProperty
+      });
 
       dispatch(
-        setViewportFeatures({
+        setViewportFeaturesReady({
```

Comment:
Doubts on naming for this... Without reading further code... what is "ready"? viewport / viewportFeatures...?

Reply:
ViewportFeaturesReady means that viewportFeatures has been calculated and are ready to be used (by widgets for example). What name do you suggest?